internal/resources/metal/virtual_circuit/resource_test.go

```diff
@@ -130,3 +130,75 @@ func TestAccMetalVirtualCircuit_dedicated(t *testing.T) {
 		},
 	})
 }
+
+func testAccMetalConnectionConfig_SharedPrimaryVrfWithVirtualCircuit(randstr string) string {
+	return fmt.Sprintf(`
+        resource "equinix_metal_project" "test" {
+            name = "tfacc-conn-pro-%s"
+        }
+
+        resource "equinix_metal_vrf" "test1" {
+            name        = "tfacc-conn-vrf1-%s"
+            metro       = "sv"
+            local_asn   = "65001"
+            ip_ranges   = ["10.0.0.0/16", "2604:1380:4641:a00::/59"]
+            project_id  = equinix_metal_project.test.id
+        }
+
+        resource "equinix_metal_connection" "test" {
+            name               = "tfacc-conn-%s"
+            project_id         = equinix_metal_project.test.id
+            type               = "shared"
+            redundancy         = "primary"
+            metro              = "sv"
+			speed              = "200Mbps"
+			service_token_type = "a_side"
+			contact_email      = "tfacc@example.com"
+			vrfs               = [
+				equinix_metal_vrf.test1.id,
+			]
+        }
+			
+		resource "equinix_metal_virtual_circuit" "test" {
+            name = "tfacc-vc-%s"
+            description = "tfacc-vc-%s"
+            connection_id = equinix_metal_connection.test.id
+			virtual_circuit_id = equinix_metal_connection.test.ports[0].virtual_circuit_ids[0]
+            project_id = equinix_metal_project.test.id
+            port_id = equinix_metal_connection.test.ports[0].id
+            vrf_id = equinix_metal_vrf.test1.id
+			subnet = "10.0.0.0/31"
+			subnet_ipv6 = "2604:1380:4641:a00::4/126"
+        }`,
+		randstr, randstr, randstr, randstr, randstr)
+}
+
+func TestAccMetalConnection_sharedVrfWithVC(t *testing.T) {
```

Comment:
Test looks good and I see it passing, but the name doesn't line up with others in the file (the name prefix should reference virtual circuit rather than connection, since the focus of the test is validating the VC resource).

```suggestion
func TestAccMetalVirtualCircuit_sharedVrf(t *testing.T) {
```